### DISCUSSION TITLE
modify nnunet

### DISCUSSION THREAD

**Discussion by mariaconcettaveraldi** [MIC-DKFZ/nnUNet#1416] [Q&A]
Hello!
I would like to know how can i modify nnunet to create a single channel architecture with a linear activation function.

Thank you,
Mariaconcetta

**Comment by pzaffino** [comment]
Hi all,
I have exactly the same problem.
I would have a custom, 3D, single-channel, nnUNet that gives as an output a 3D volume with a linear activation function (I do not want a segmentation).
Any suggestion on how to implement this?

Best,
Paolo

**Comment by FabianIsensee** [comment]
Hey, nnU-Net is build for segmentation. It cannot do other things. If all you need is the same UNet then please take a look at this function: 
https://github.com/MIC-DKFZ/nnUNet/blob/1635a5ec2ce758c82f2593732a67956f2cf9a2ff/nnunetv2/training/nnUNetTrainer/nnUNetTrainer.py#L258

Best,
Fabian

**Reply by pzaffino** [reply]
Thank you @FabianIsensee 
I understood.

**Comment by ibro45** [comment]
Hi @pzaffino @mariaconcettaveraldi have you had any luck with it? I can't seem to successfully train a custom model https://github.com/MIC-DKFZ/nnUNet/issues/1479.

**Reply by pzaffino** [reply]
Hi,
no, I just switched to another architecture.
I am interested in regression rather than segmentation, and nnunet is specifically built for segmentation.
